tasks/main.yml

```
@@ -444,30 +449,42 @@
     state: present
   when: __mssql_tuned_supported | bool
 
-- name: Ensure that the tuned service is started and enabled
+- name: Ensure that the tuned service is enabled
   service:
     name: tuned
-    state: started
     enabled: true
   when: __mssql_tuned_supported | bool
 
+- name: Ensure that the tuned service is started
+  service:
+    name: tuned
+    state: started
+  when:
+    - __mssql_is_booted
```

```suggestion
    - __mssql_is_booted | bool
```
Not sure if this is needed anymore - it used to be required - but it also makes it consistent with the line below

---

I didn't `| bool`ify that flag anywhere else, nor in any other role. There's no need to, it's already a boolean, and if it isn't, we messed up and want to know (i.e. tests fail).

`__mssql_tuned_supported` is suspect indeed: there is *no* default value in vars/main.yml or anywhere else, just these two special cases:
```
vars/RedHat.yml:__mssql_tuned_supported: true
vars/Suse.yml:__mssql_tuned_supported: false
```

So for e.g. Fedora this comes out as `undefined | bool`, which I assume is `false` instead of an error. I'd prefer to clean this up and add `__mssql_tuned_supported: false` to vars/main.yml.

I sent PR #354 to clean this up.

---

> I didn't `| bool`ify that flag anywhere else, nor in any other role. There's no need to, it's already a boolean, and if it isn't, we messed up and want to know (i.e. tests fail).

The `| bool` (or similar) used to be required because of the Ansible bare variable issue - https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_2.8.html#bare-variables-in-conditionals - which is still a thing in ansible 2.9 - so we *might* still need it when running on 2.9 - the reason why ansible-lint, ansible-test, etc. do not warn is because they don't care about ansible 2.15 or earlier.

> 
> `__mssql_tuned_supported` is suspect indeed: there is _no_ default value in vars/main.yml or anywhere else, just these two special cases:
> 
> ```
> vars/RedHat.yml:__mssql_tuned_supported: true
> vars/Suse.yml:__mssql_tuned_supported: false
> ```
> 
> So for e.g. Fedora this comes out as `undefined | bool`, which I assume is `false` instead of an error.

It is defined on Fedora via the RedHat.yml vars file.  Because of the way this logic works - https://github.com/linux-system-roles/mssql/blob/main/tasks/set_vars.yml#L9 - `ansible_os_family == "RedHat"` on Fedora (and CentOS, Rocky, Alma, etc.)

>  I'd prefer to clean this up and add `__mssql_tuned_supported: false` to vars/main.yml.

And that's fine, but it won't do anything in this case.

> 
> I sent PR #354 to clean this up.

---

OK -- didn't seem to break the C7 tests in the other projects, but I just added `| bool` everywhere.